src/fu-usb-backend.c

```diff
@@ -20,6 +20,33 @@ struct _FuUsbBackend {
 
 G_DEFINE_TYPE(FuUsbBackend, fu_usb_backend, FU_TYPE_BACKEND)
 
+#define FU_USB_BACKEND_POLL_INTERVAL_DEFAULT	 1000 /* ms */
+#define FU_USB_BACKEND_POLL_INTERVAL_WAIT_REPLUG 5 /* ms */
+
+#ifdef _WIN32
+static void
+fu_usb_backend_device_notify_flags_cb(FuDevice *device, GParamSpec *pspec, FuBackend *backend)
+{
+#if G_USB_CHECK_VERSION(0, 3, 10)
+	FuUsbBackend *self = FU_USB_BACKEND(backend);
+
+	/* if waiting for a disconnect, set win32 to poll insanely fast -- and set it
+	 * back to the default when the device removal was detected */
+	if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG)) {
+		g_debug("setting USB poll interval to %ums to detect replug",
+			(guint)FU_USB_BACKEND_POLL_INTERVAL_WAIT_REPLUG);
+		g_usb_context_set_hotplug_poll_interval(self->usb_ctx,
+							FU_USB_BACKEND_POLL_INTERVAL_WAIT_REPLUG);
+	} else {
+		g_usb_context_set_hotplug_poll_interval(self->usb_ctx,
+							FU_USB_BACKEND_POLL_INTERVAL_DEFAULT);
+	}
+#else
+	g_warning("GUsb >= 0.3.10 may be needed to notice device enumeration");
```

Comment:
this is going to be needlessly noisy for non-Win32.

Reply:
Oh nvm- nested `#ifdef`, I missed it.

Reply:
Ohh the entire function is in `#ifdef _WIN32`